### PR TITLE
Test with docker

### PR DESCRIPTION
**Pull request by leohemsted**
Add test-with-docker and integration-test-with-docker commands to make local development easier.

nb: uses the dockerfile we originally created for concourse (that has been postponed due to difficulties publishing/releasing from a linux docker container). 

Uses different commands to appveyor, which runs the commands written in appveyor.txt. I think this is fine though.

Sample output:
```
$ make integration-test-with-docker
scripts/generate_docker_env.sh
docker build -t govuk/notify-net-client-builder .
Sending build context to Docker daemon  4.326MB
Step 1/3 : FROM mcr.microsoft.com/dotnet/core/sdk:2.1
 ---> 7eb1ecf4a018
Step 2/3 : RUN     echo "Install base packages"     && apt-get update     && apt-get install -y --no-install-recommends     gnupg     make
 ---> Using cache
 ---> 66ed77c275ce
Step 3/3 : WORKDIR /var/project
 ---> Using cache
 ---> fb15168ff68c
Successfully built fb15168ff68c
Successfully tagged govuk/notify-net-client-builder:latest
docker run -it --rm \
		--name "leohemsted-notifications-net-client-manual-test" \
		-v "`pwd`:/var/project" \
		--env-file docker.env \
		govuk/notify-net-client-builder \
		make build-test
dotnet build -f=netcoreapp2.0
Microsoft (R) Build Engine version 16.2.32702+c4012a063 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 2.68 sec for /var/project/src/Notify.Tests/Notify.Tests.csproj.
  Restore completed in 2.68 sec for /var/project/src/Notify/Notify.csproj.
  Notify -> /var/project/src/Notify/bin/Debug/netcoreapp2.0/Notify.dll
  Notify.Tests -> /var/project/src/Notify.Tests/bin/Debug/netcoreapp2.0/Notify.Tests.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.18
dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Unit
Test run for /var/project/src/Notify.Tests/bin/Debug/netcoreapp2.0/Notify.Tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 16.2.0-preview-20190606-02
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Test Run Successful.
Total tests: 86
     Passed: 86
 Total time: 2.0698 Seconds
dotnet test ./src/Notify.Tests/Notify.Tests.csproj --no-build -f=netcoreapp2.0 --filter TestCategory=Integration
Test run for /var/project/src/Notify.Tests/bin/Debug/netcoreapp2.0/Notify.Tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 16.2.0-preview-20190606-02
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Test Run Successful.
Total tests: 56
     Passed: 56
 Total time: 13.5587 Seconds
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,17 +11,26 @@ If you are not working on a windows OS, .Net Frameworks are not supported but yo
 
 Before running tests please ensure that the environment variables are set up for the test project.
 
+```sh
```
Comment (idavidmcdonald) [inline CONTRIBUTING.md]: I think all these variables are only needed when running the integration tests? Maybe we could make that clearer in the readme like

### Unit tests

Run `make test-with-docker`

### Integration tests

export the stuff and run make ...

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,7 @@
 .DEFAULT_GOAL := help
 
+DOCKER_BUILDER_IMAGE_NAME = govuk/notify-net-client-builder
```
Comment (idavidmcdonald) [inline Makefile]: Our images are now given slightly better names, you can see them in https://hub.docker.com/u/govuknotify

So this one should be `govuknotify/net-client-tests`

File: Makefile [hunk 3]
```diff
@@ -8,53 +10,66 @@ help:
 build: ## Build project
 	dotnet build -f=netcoreapp2.0
 
-.PHONY: test
```
Comment (idavidmcdonald) [inline Makefile]: I think we still want to keep `test` as this keeps a common interface with all our other clients. We use this interface in concourse - https://github.com/alphagov/notifications-aws/blob/master/concourse/client-library-pipeline.yml.j2#L193

File: Makefile [hunk 4]
```diff
@@ -8,53 +10,66 @@ help:
...
+
+.PHONY: test-with-docker
+test-with-docker: generate-env-file ## Test with docker
+	docker build -t ${DOCKER_BUILDER_IMAGE_NAME} .
+	docker run -it --rm \
+		--name "${USER}-notifications-net-client-manual-test" \
+		-v "`pwd`:/var/project" \
+		--env-file docker.env \
+		${DOCKER_BUILDER_IMAGE_NAME} \
+		make build-unit-test
```
Comment (idavidmcdonald) [inline Makefile]: We've got two slightly conflicting principles in this make file

In this one, make `test-with-docker` will just run the unit tests but further up we are explicit like `make unit-test`. We also have `build-test` which will run both unit and integration.

Personally, I prefer `make test` to just run unit tests and `make integration-tests` to just run integration tests and this fits with what concourse is doing I feel. I don't feel too strongly which but I feel a bit more consistency could be good.

Comment (leohemsted) [reply Makefile]: sensible, changed to reflect yr naming. also removed some more stuff

**Pull request merged**